firmware/user/modes/mode_tunernome.c

```diff
@@ -0,0 +1,721 @@
+{
+    return getMagnitude(idx + 1) - getMagnitude(idx - 1);
+}
+
+/**
+ * Inline helper function to get the magnitude of a frequency bin from folded_bins[]
+ *
+ * @param idx The index to get a magnitude from
+ * @return A signed magnitude, even though folded_bins[] is unsigned
+ */
+static inline int16_t getSemiMagnitude(int16_t idx)
+{
+    if(idx < 0)
+    {
+        idx += FIXBPERO;
+    }
+    if(idx > FIXBPERO - 1)
+    {
+        idx -= FIXBPERO;
+    }
+    return folded_bins[idx];
+}
+
+/**
+ * Inline helper function to get the difference in magnitudes around a given
+ * frequency bin from folded_bins[]
+ *
+ * @param idx The index to get a difference in magnitudes around
+ * @return The difference in magnitudes of the bins before and after the given index
+ */
+static inline int16_t getSemiDiffAround(uint16_t idx)
+{
+    return getSemiMagnitude(idx + 1) - getSemiMagnitude(idx - 1);
+}
+
+/**
+ * Recalculate the per-bpm values for the metronome
+ */
+void ICACHE_FLASH_ATTR recalcMetronome() {
+    tunernome->bps = tunernome->bpm / 60.0f;
+    tunernome->periodS = 1.0f / tunernome->bps;
+    tunernome->periodMS = 1000.0f * tunernome->periodS;
+    tunernome->cycleFrames = tunernome->periodMS / TUNERNOME_UPDATE_MS;
+    tunernome->halfCycleFrames = tunernome->cycleFrames * 0.5f;
+    tunernome->tockFrame1 = round(tunernome->cycleFrames * 0.25f);
+    tunernome->tockFrame2 = round(tunernome->cycleFrames * 0.75f);
+    tunernome->finalCycleFrame = round(tunernome->cycleFrames);
+}
+
+/**
+ * @brief called on a timer, updates the game state
+ *
+ * @param arg
+ */
+static void ICACHE_FLASH_ATTR tunernomeUpdate(void* arg __attribute__((unused)))
+{
+    switch(tunernome->mode)
+    {
+        default:
+        case TN_TUNER:
+        {
+            plotText(0, OLED_HEIGHT - FONT_HEIGHT_TOMTHUMB, "< Exit", TOM_THUMB, WHITE);
+            plotText(OLED_WIDTH - 43, OLED_HEIGHT - FONT_HEIGHT_TOMTHUMB, "Metronome >", TOM_THUMB, WHITE);
+            break;
+        }
+        case TN_METRONOME:
+        {
+            clearDisplay();
+
+            if(!tunernome->pause)
+            {
+                // TODO: calculate x and y
+                
+                float intermedX = -1 * cosf(tunernome->frame * M_PI / tunernome->halfCycleFrames );
+                float intermedY = -1 * sinf(tunernome->frame * M_PI / tunernome->halfCycleFrames );
+
+                char tempStr[50] = {0};
+                ets_sprintf(tempStr, "%d.%d", (int)(tunernome->frame /** M_PI / tunernome->halfCycleFrames*/), (int)((tunernome->frame /** M_PI / tunernome->halfCycleFrames*/)*1000)%1000);
+                plotText(0, 0, tempStr, IBM_VGA_8, WHITE);
+                int x = round(METRONOME_CENTER_X - (intermedX * METRONOME_RADIUS));
+                int y = round(METRONOME_CENTER_Y - (ABS(intermedY) * METRONOME_RADIUS));
+
+                if(tunernome->frame == tunernome->tockFrame1 || tunernome->frame == tunernome->tockFrame2)
```

Comment:
You should blink the LEDs when the arm is most horizontal, not most vertical, like the tocks here: https://www.youtube.com/watch?v=Hgc7pA43DgI

Reply:
Comparing the metronome here to that video too, I think yours is running at 2x speed. Gotta go fast!

Reply:
https://www.youtube.com/watch?v=gsJEMH_emBM

You are correct about the double speed, though!

Reply:
Huh, two videos, two behaviors. Lets ask the actual musicians what it should be.

Reply:
pf3k has spoken! Blinks at the ends of the swing arc.

Reply:
It has been done

Reply:
Has it? You should push your changes if you've made 'em.

Reply:
Also accept the PR from my master to your branch to fix the OLED flashing issues
https://github.com/Brycey92/Super-2021-Swadge-FW-Sandbox/pull/3